src/Symfony/Component/Console/CommandLoader/ContainerCommandLoader.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Symfony\Component\Console\CommandLoader;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+/**
+ * Loads commands from a PSR-11 container.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class ContainerCommandLoader implements CommandLoaderInterface
+{
+    private $container;
+    private $commandMap;
+
+    /**
+     * @param ContainerInterface $container  A container from which to load command services
+     * @param array              $commandMap An array with command names as keys and service ids as values
+     */
+    public function __construct(ContainerInterface $container, array $commandMap)
```

Comment:
I started with only command names, but @mnapoli noticed that it prevents mapping a command to a service that is not named the same as the command (https://github.com/symfony/symfony/pull/22734#discussion_r125921985), which can be useful especially outside of the fullstack I think. 
A good side effect is that using a `commandMap` makes that we register only one closure for a command and its aliases (they're all mapped to the same service id in the commandMap array instead of in the locator factories, that gives a lighter container). Still not convinced?